integrations/axum/src/lib.rs

```diff
@@ -110,20 +110,54 @@ pub fn redirect(cx: leptos::Scope, path: &str) {
 }
 
 /// Decomposes an HTTP request into its parts, allowing you to read its headers
-/// and other data without consuming the body.
-pub async fn generate_request_parts(req: Request<Body>) -> RequestParts {
+/// and other data without consuming the body. Creates a new Request from the
+/// original parts for further processsing
+pub async fn generate_request_parts(
```

Comment:
I'd suggest renaming this to `generate_request_and_parts`, leaving the original `generate_request_parts` alongside it (deprecated), and using the new `generate_request_and_parts`. This would allow us to merge this without it being a breaking change for semver purposes, even if nobody is actually using `generate_request_parts` outside the integration.